tiled/server/app.py

```diff
@@ -848,16 +867,27 @@ async def capture_metrics_prometheus(request: Request, call_next):
             try:
                 response = await call_next(request)
             except Exception:
-                # Make a placeholder to feed to capture_request_metrics.
-                # The actual error response will be created by FastAPI/Starlette
-                # further up the call stack, where this exception will propagate.
+                # Make a placeholder response to feed to capture_request_metrics.
+                # Notice that this placeholder response will under no circumstances
+                # be returned below. It is fed to capture_request_metrics,
+                # and then the caught exception is re-raised.
+                #
+                # The re-raised exception propagates to http_exception_handler,
+                # which creates the Response that will actually be sent to the
+                # client, complete with some useful headers.
                 response = Response(status_code=500)
                 raise
             finally:
                 response.__class__ = PatchedStreamingResponse  # tolerate memoryview
                 metrics.capture_request_metrics(request, response)
             return response
```

Comment:
It took me while to catch the meaning of the comments.  Is this succinct version clearer?

```suggestion
                # Make an ephemeral response solely for 'capture_request_metrics'.
                # It will only be used in the 'finally' clean-up block, then goes out of scope
                only_for_metrics = Response(status_code=500)
                response = only_for_metrics
                # Now re-raise the exception so that the server can generate and send
                # an appropriate response to the client
                raise
            finally:
                response.__class__ = PatchedStreamingResponse  # tolerate memoryview
                metrics.capture_request_metrics(request, response)

            # The ephemeral 'only_for_metrics' response has already been discarded
            return response
```